### ISSUE TITLE
`@stylistic/indent`: ts return type indentation broken as of 5.x

### ISSUE THREAD

**Issue by johncrim** [eslint-stylistic/eslint-stylistic#875]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Given correctly formatted TS code with return type info on the next line (b/c the parameter list is long or return type is long), eg:

```ts
export function isAuthenticated(authResult: AuthenticationResult | null | undefined)
  : authResult is SuccessAuthenticationResult {
  return !! authResult && authResult.isAuthenticated();
}
```

@stylistic/eslint-plugin 4.4.1 reports no lint error (correct)

@stylistic/eslint-plugin 5.x and 5.1.0 reports:
```
  112:1  warning  Expected indentation of 0 spaces but found 2  @stylistic/indent
```

This fix did not fix the issue:
https://github.com/eslint-stylistic/eslint-stylistic/pull/853

Related but different new issues on the same rule:
https://github.com/eslint-stylistic/eslint-stylistic/issues/845
https://github.com/eslint-stylistic/eslint-stylistic/issues/851

### Reproduction

https://stackblitz.com/edit/github-c4wdzgzf

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by johncrim** [comment]
Note: The ability to set the indentation for declaration line continuations would be great!  Eg I would prefer an extra level of indentation for line continuations.

**Comment by 9romise** [comment]
Sounds reasonable.
And I think a `returnType` option for `FunctionExpression` and `FunctionDeclaration` would be a good solution.
PR welcome!
